src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -48,6 +51,16 @@ public User addBookToUser (@PathVariable Long id, @RequestBody Book bookToAdd) t
         return userService.addBook(id, bookToAdd);
     }
 
+    @PutMapping("/{userId}/books/{bookId}")
+    public User addBookById(@PathVariable Long userId, @PathVariable Long bookId) {
+         Optional<User> user = userRepository.findById(userId);
```

Comment:
You should do this as a findById().orElseThrow(UserNotFoundException) and then the addBookById().